app/controllers/proposals_controller.rb

```diff
@@ -1,10 +1,11 @@
 class ProposalsController < ApplicationController
+  skip_authorization_check
```

Comment:
We should not have that, because we do want other actions to get authorized. You need to remove this line and add an ability for the new action `:toggle_favorite`

Once you have that in ability, you can either let load_and_authorize_resource work its magic, or you manually authorize your action.